lab_1/main.py

```diff
@@ -74,7 +246,36 @@ def detect_language(unknown_profile: dict, profile_1: dict, profile_2: dict, top
     :param top_n: a number of the most common words
     :return: a language
     """
-    pass
+
+    if not isinstance(unknown_profile, dict) or not isinstance(profile_1, dict) \
+            or not isinstance(profile_2, dict) or not isinstance(top_n, int):
+        return None
+
+    if score_en > score_de:
+        unknown_profile['name'] = profile_1['name']
+        language_unknown = language_en
+        return language_unknown
+    if score_en < score_de:
+        unknown_profile['name'] = profile_2['name']
+        language_unknown = language_de
+        return language_unknown
+    else:
+        lang = [language_en, language_de]
+        prof = {profile_1: language_en, profile_2: language_de}
+        lang.sort()
+        for i in prof:
+            if i == lang[0]:
+                language_unknown = i
+                return language_unknown
+
+score_en = compare_profiles(unknown_profile, en_profile, top_n_en)
+score_de = compare_profiles(unknown_profile, de_profile, top_n_de)
+
+en_profile = create_language_profile(language_en, en_text, en_stop_words)
+de_profile = create_language_profile(language_de, de_text, de_stop_words)
+unknown_profile = create_language_profile(language_unknown, unknown_text, unknown_stop_words)
+
+top_n_unknown = len(get_top_n_words(freq_dict_unknown, top_n_unknown))
```

Comment:
The `main.py` file should not contain function calls - you should check your implementation using the `start.py` file.